app/code/Magento/CheckoutAgreements/Model/AgreementsConfigProvider.php

```diff
@@ -29,6 +29,11 @@ class AgreementsConfigProvider implements ConfigProviderInterface
      * @var \Magento\Framework\Escaper
      */
     protected $escaper;
+    
+    /**
+     * @var \Magento\Widget\Model\Template\Filter
+     */
+    protected $templateProcessor;
```

Comment:
Please change the access level to protected and add this new dependency in a backward compatible way as mentioned in comments above